scripts/fn_paradrop.sqf

```diff
@@ -0,0 +1,198 @@
+
+//expand waypoint 
+private _dp = _pilotGroup addWaypoint [_dropZoneLocs select 1,-1,currentWaypoint _pilotGroup + 1,""];
+if(!_disableWP2) then {
+    private _de = _pilotGroup addWaypoint [_dropZoneLocs select 0,-1,currentWaypoint _pilotGroup + 2,""];
+};
+
+// Add the waypoint paradrop operation.  Eject automatically gives people a parachute 
+// Adds a key,value pair to the vehicle characteristics hash that can be referenced globally
+// No global variable names required
+_vehicle setVariable ['tac_dropGroups',_dropGroups];
+_vehicle setVariable ['tac_chuteHeight',_chuteHeight];
+
+// note - moveOut instantly pushes them out of vehicle without playing animation
+// this emulates a static line drop dropping very quickly, but needs to be spaced
+// it also means that you need to spawn in a parachute otherwise they freefall to their deaths
+// currently this is set for a drop interval of 1/2 second, allowing AI to keep cohesion
+// but not collide into each other
+
+// if  _groupAjudst is not used, then each group being parachuted will getOut a unit
+// at the same time as the forEach executes in parallel. 
+// Nominally, could use count units _x with _forEachIndex but will not work with groups
+// of different sizes
+
+//  Other notes:
+//  Vehicles are unassigned and allowGetIn is set to false to prevent the paratroopers from
+//  running back to a landed plane or helicopter to get back in again.
+//  Velocity adjustment occurs to differentiate helicopters from planes to alter jump intervals
+
+private _fnc_dropParas = {
+    private _vehicle = vehicle this; 
+    private _paraGroups = _vehicle getVariable ['tac_dropGroups', []];
+    private _chuteHeight = _vehicle getVariable ['tac_chuteHeight',[]];
+    private _dropCount = _paraGroups apply {count units _x};
+    private _groupAdjust = [];
+    _groupAdjust resize (count _dropCount);
+
+    for "_i" from 0 to (count _dropCount)-1 do {
+        private _sum = 0;
+        for "_j" from 0 to _i do {
+            _sum = _sum + (_dropCount select _j);
+        };
+        _groupAdjust set [_i,_sum];
+    };
+
+    private _jumpFreq = 0.3;
+    if(speed _vehicle < 250) then {_jumpFreq = 0.5;};
+
+    {
+        units _x allowGetIn false;
+        private _groupCount = _forEachIndex;
+        {
+            private _jumpDelay = _forEachIndex*_jumpFreq+(_groupAdjust select _groupCount)*_jumpFreq;
+            [{
+                private _man = _this select 0;
+                private _chuteHeight = _this select 1;
+                unassignVehicle _man;
+                moveOut _man;
+                [_man, _chuteHeight] spawn {
+                    params ['_man','_chuteHeight'];
+                    waitUntil {sleep 0.5; ((getPosATL _man) select 2) <= _chuteHeight};
+                    _chute = createVehicle ['Steerable_Parachute_F', position _man, [], 0, 'FLY'];
+                    _chute setPos (getPosATL _man);
+                    _man moveInDriver _chute;
+                };
```

Comment:
I assume grabbing and storing the ATL/ASL position is to prevent changes in position from causing issues during execution?  As before, I wasn't aware of `CBA_fnc_waitUntilAndExecute`, so I'll use that in the future.  

One issue that I see here @jonpas is that I understand why you'd put the `{_man distance _vehicle >= 15}` condition in to avoid collision with the vehicle, but it does mean I have to do some fairly extensive injury testing on the people being dropped.  This makes the minimum height for parachute deployment about 75m using this script, which is unrealistically low.  Could be that Arma just doesn't represent the deceleration very well in which case who cares, but otherwise it's a problem.

Reply:
Distance check changes it because `_man` exits `_vehicle` at the vehicle's current speed, meaning that opening up the distance depends at first mostly on Z-directional acceleration.  75m is the preset option 2 ("dangerous drop") height of 90m - 15m, so in fairness it is an exaggeration to say that it will make it 75m.  However, it does present a potential issue in terms of parachute deployment being much lower than we want it, strictly speaking.

One alternative way of resolving this I've seen is just to do something like `_man setVelocity [0,0,5]` right after the `getOut _man` execution which solves the distance issue as well, and makes drop zones a more accurate reflection of where troops land, but if it's ever used by players and not AI they'll have a... "jerky" experience and I chose not to do that.  

Ultimately I think what we have to do is retain your solution, adjust the height of the "dangerous drop" preset upwards and call it a day.  It's based on the Rhodesian C-47 drops which were typically around 90m and as low as 70m even though officially they were supposed to be around 120m; however, it won't be that much different at 120m.  Also, I don't expect it to be used much, as it's "dangerous" for a reason, the Rhodesians took 10% casualties from every drop just by virtue of landing.

Reply:
Soldiers do not seem to retain velocity, or they lose it really really fast. I don't think it's an issue, but feel free to adapt however to make it work nicely.